### ISSUE TITLE
Can't Load Swagger Bundle

### ISSUE THREAD

**Issue by Redoubts** [swagger-api/swagger-ui#4038]
I'm trying the new 3.7.0 tagged release, by replacing my assets with the compiled files under /dist/. When I try this, I see `[Error] ReferenceError: Can't find variable: SwaggerUIBundle
	onload (api_docs.html:106)`, preceded by this error trace:

```
[Error] TypeError: Argument 1 ('element') to Window.getComputedStyle must be an instance of Element
	getComputedStyle (swagger-ui-bundle.js:98:73714)
	e (swagger-ui-bundle.js:98:73714)
	(anonymous function) (swagger-ui-bundle.js:98:75466)
	(anonymous function) (swagger-ui-bundle.js:98:73533)
	(anonymous function) (swagger-ui-bundle.js:98:73600)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:52:43854)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:52:42891)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:62:127389)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:62:125540)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:53:96474)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:98:79641)
	t (swagger-ui-bundle.js:1:339)
	(anonymous function) (swagger-ui-bundle.js:1:724)
	(anonymous function) (swagger-ui-bundle.js:1:735)
	(anonymous function) (swagger-ui-bundle.js:1:209)
	Global Code (swagger-ui-bundle.js:1:212
```

**Comment by shockey** [comment]
@Redoubts, what browser are you using?

**Comment by Redoubts** [comment]
This error is from Safari, but I saw a similar problem in Chrome.

**Comment by shockey** [comment]
Thanks @Redoubts!

So, this is coming from the Zenscroll library, which we added in 3.7.0. By sheer luck, I happened to get the error you mentioned while trying to integrate Zenscroll in a totally unrelated project of mine, and reported the issue: https://github.com/zengabor/zenscroll/issues/50

When it's imported, it runs some startup code, which IMO is questionable. It relies on `document.body` being available, which is all good, unless you're pulling in the JS within your `<head>` (you are!), which clearly is parsed before `document.body` is available.

So, you have a couple of options here:
- Add the `async` attribute to your script tags, so it doesn't block HTML parsing (and therefore `document.body` availability)
- Add `window.noZensmooth = true` in a script tag _above_ where you're pulling in Swagger-UI, to avoid execution of the problematic Zenscroll startup code

I'm not super happy that this is happening... strictly speaking, this is a breaking bug for folks that use script-in-head implementations, even though it's not in our codebase.

I'm hoping to have the time to make a PR against Zenscroll this week - though it looks like a fork might be a better option, since the project is pretty inactive.

In the meantime, let me know if my proposed solutions don't work for you!

**Comment by Redoubts** [comment]
Thanks for all the helpful options. I’ll take a look.

**Comment by shockey** [comment]
Closing due to inactivity. 

This is simply to keep our issue tracker clean - feel free to comment if there are any further thoughts or concerns, and we'll be happy to reopen this issue.

**Comment by jfhumann** [comment]
What's the status of this? Would like to update to a newer version of swagger-ui, but this error still blocks it.

**Comment by shockey** [comment]
Hi @jfhumann! No major updates - this isn't something that's actionable for us, as the issue exists in a dependency library, not Swagger-UI (see https://github.com/zengabor/zenscroll/issues/50).

If you need this ironed out, consider submitting a pull request to Zenscroll, bumping the discussion on the issue over there, or using one of the workarounds I mentioned in my previous comment.

If there's anything else I can help with, don't hesitate to reply.

**Comment by shockey** [comment]
Hi folks - Zenscroll has released `zenscroll@4.0.1` which fixes this issue on their end.

Future versions of Swagger-UI will capture this fix as well.
